### PR TITLE
Use modified paraswap package version

### LINKED ISSUE #2449: Faster performance of UI: Reduce dependencies

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Fixes #2449

- updated paraswap package to use custom modified version where the web3 is not included
- also added web3 and some related dependencies as dev dependencies because dex-js

Here are the changes I made in Paraswap package https://github.com/paraswap/paraswap-sdk/pull/109/files
I published this as a npm module with my account
The result is reduction of total bundle from All (6.28 MB) to All (4.93 MB)

Before
screencapture-127-0-0-1-8888-2022-02-24-11_53_21

After
screencapture-127-0-0-1-8888-2022-02-24-11_54_23

# To test
- Mainly we should make sure that the paraswap requests work as expected and there is nothing weird with price fetching

### CODE REVIEW

**Review by anxolin — COMMENTED** [review]
Unrelated PR, but i see also a lot of ethersjs code that doesn't get tree shaken.
Maybe this help: 

https://twitter.com/DawsonBotsford/status/1394048094338129922?s=20&t=8MXUOAJOdaijh_uX0cohog


The proposed NPM package, is nice (if it works), although i wonder if u could make a version that Paraswap would be willing to merge. 
Maybe one solution is to:
- Not delete the web3 dependency, mark it as peer or optional (not sure)
- import only the specific things that are required
- I think they probably shouldn't instanciate the web3 object there, make it optional

Other approach, if the previous is not easily done, is to see what do we need from them. Cause it feels their library adds too many dependencies.
I'll raise also this to them to see if they know how to get around it, but loading all lodash, bignumber and axios seems like too much for just doing a simple request

**Comment by anxolin** [comment]
@nenadV91 I've open this discussion, i think Paraswap library should have less dependencies: https://gnosisinc.slack.com/archives/C0203Q9R4JJ/p1645723122926249

File: package.json [hunk 1]
```diff
@@ -150,6 +150,9 @@
     "use-count-up": "^2.2.5",
     "wcag-contrast": "^3.0.0",
     "web-vitals": "^2.1.0",
+    "web3": "^1.7.0",
+    "web3-providers-http": "^1.7.0",
+    "web3-providers-ws": "^1.7.0",
```
[the remark this answers is not in the document]
Comment (anxolin) [reply package.json:155]: Ahh, because is peer dependency... okok
maybe should be optional in dex-js (since we don't use it), but fine

**Comment by elena-zh** [comment]
Hey @nenadV91 , I always get 403 error response  on each paraswap request in this PR
image

However, I get 403 error on Prod as well. Weird. 

Nevertheless, I have not noticed any issues in price fetching in this PR.

**Comment by nenadV91** [comment]
@elena-zh For me Paraswap requests work fine both in this PR and on prod.

**Comment by elena-zh** [comment]
@nenadV91 , seems that an access to paraswap.io is totally blocked for me 
image

File: package.json [hunk 2]
```diff
@@ -225,7 +228,7 @@
     "fast-safe-stringify": "^2.0.8",
     "firebase": "^9.1.3",
     "ipfs-http-client": "^52.0.3",
-    "paraswap": "^5.0.1",
+    "paraswap": "npm:@nenad91/paraswap#5.1.0",
```
Comment (W3stside) [inline package.json:231]: cool!

**Review by W3stside — APPROVED** [review]

**Review by anxolin — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Using VPN, I was able to test the changes. LGTM!

**Pull request merged**